test/operators/test_fermionic_op.py

```diff
@@ -0,0 +1,159 @@
+        self.assertEqual(fer_op.to_list(), [("+-", 2)])
+
+        fer_op = (2 + 1j) * FermionicOp([("+N", 3), ("E-", 1)])
+        self.assertEqual(fer_op.to_list(), [("+N", (6 + 3j)), ("E-", (2 + 1j))])
+
+    def test_div(self):
+        """Test __truediv__"""
+        fer_op = FermionicOp([("+N", 3), ("E-", 1)]) / 3
+        self.assertEqual(fer_op.to_list(), [("+N", 1.0), ("E-", 0.3333333333333333)])
+
+    def test_add(self):
+        """Test __add__"""
+        fer_op = 3 * FermionicOp("+N") + FermionicOp("E-")
+        self.assertListEqual(fer_op.to_list(), [("+N", 3), ("E-", 1)])
+
+        fer_op = sum(FermionicOp(label) for label in ['NIII', 'INII', 'IINI', 'IIIN'])
+        self.assertListEqual(fer_op.to_list(), [('NIII', 1), ('INII', 1), ('IINI', 1), ('IIIN', 1)])
+
+    def test_sub(self):
+        """Test __sub__"""
+        fer_op = 3 * FermionicOp("++") - 2 * FermionicOp("--")
+        self.assertListEqual(fer_op.to_list(), [("++", 3), ("--", -2)])
+
+    @data(*product(fermion_labels(1), fermion_labels(1)))
+    @unpack
+    def test_matmul(self, label1, label2):
+        """Test matrix multiplication"""
+        fer_op = FermionicOp(label1) @ FermionicOp(label2)
+        mapping = {
+            "II": "I",
+            "I+": "+",
+            "I-": "-",
+            "IN": "N",
+            "IE": "E",
+            "+I": "+",
+            "++": 0,
+            "+-": "N",
+            "+N": 0,
+            "+E": "+",
+            "-I": "-",
+            "-+": "E",
+            "--": 0,
+            "-N": "-",
+            "-E": 0,
+            "NI": "N",
+            "N+": "+",
+            "N-": 0,
+            "NN": "N",
+            "NE": 0,
+            "EI": "E",
+            "E+": 0,
+            "E-": "-",
+            "EN": 0,
+            "EE": "E",
+        }
+        result = mapping[label1 + label2]
+        expected = [(result, 1)] if result != 0 else [("I", 0)]
+        self.assertListEqual(fer_op.to_list(), expected)
+
+    def test_matmul_multi(self):
+        """Test matrix multiplication"""
+        fer_op = FermionicOp("+-") @ FermionicOp("-I")
+        self.assertListEqual(fer_op.to_list(), [("N-", -1)])
+
+        fer_op = (FermionicOp("+N") + FermionicOp("E-")) @ (
+            FermionicOp("II") + FermionicOp("-+")
+        )
+        self.assertListEqual(
+            fer_op.to_list(), [("+N", 1), ("N+", 1), ("E-", 1), ("-E", -1)]
+        )
+
+    def test_pow(self):
+        """Test __pow__"""
+        fer_op = FermionicOp([("+N", 3), ("E-", 1)]) ** 2
+        self.assertListEqual(fer_op.to_list(), [("II", 0)])
+
+        fer_op = FermionicOp([("+N", 3), ("N-", 1)]) ** 2
+        self.assertListEqual(fer_op.to_list(), [("+-", -3)])
+
+        fer_op = (3 * FermionicOp("IIII")) ** 3
+        self.assertListEqual(fer_op.to_list(), [("IIII", 27)])
+
+        fer_op = FermionicOp([("+N", 3), ("E-", 1)]) ** 0
+        self.assertListEqual(fer_op.to_list(), [("II", 1)])
+
+    def test_dagger(self):
```

Comment:
I think this should rather test `adjoint` directly and have another special case which asserts that `dagger` works too.